File: query_primer/query_primer__square_division_bucket.cs
using System.Linq;
using static System.Console;
using static System.Math;

internal class Program
{
    static void Main()
    {
        const int num = 10000;
        int rem = num;
        int sq = (int)Sqrt( num );

        foreach( int _ in Enumerable.Range( 0, (int)Ceiling( (decimal)num / sq ) ) ) {
            int[] a = new int[Min( rem, sq )];

            foreach( int i in Enumerable.Range( 0, Min( rem, sq ) ) ) {
                a[i] = int.Parse( ReadLine() );
                rem--;
            }
            WriteLine( a.Max() );
        }
        return;
    }
}

File: query_primer/query_primer__square_division_bucket.py
import math

N = 10000
r = N
s = math.floor(math.sqrt(N))

for _ in range(math.ceil(N / s)):
    a = []

    for i in range(min(r, s)):
        a.append(int(input()))
        r -= 1
    print(max(a))
# 実行時間 30ms
